libc-top-half/musl/arch/wasm32/pthread_arch.h

```diff
@@ -1,12 +1,11 @@
-#ifdef _REENTRANT
-#error "multiple threads not supported in musl yet"
+uintptr_t __get_tp(void) {
+#if _REENTRANT
+  int val;
+  __asm__(".globaltype g_needs_dynamic_alloc, i32\n"
+          "global.get __wasi_libc_pthread_self\n"
```

Comment:
Where are `g_needs_dynamic_alloc` and `__wasi_libc_pthread_self` defined?

As a minor style nit, wasi-libc usually uses identifiers named `__wasilibc_*` rather than `__wasi_libc_*`.